src/components/staking/ModalWithdrawRewards.vue

```diff
@@ -0,0 +1,82 @@
+<template>
+  <ActionModal
+    id="modal-withdraw-rewards"
+    ref="actionModal"
+    :submit-fn="submitForm"
+    :simulate-fn="simulateForm"
+    title="Withdraw"
+    class="modal-withdraw-rewards"
+    submission-error-prefix="Withdrawal failed"
+  >
+    <TmFormGroup
+      class="action-modal-form-group"
+      field-id="amount"
+      field-label="Amount"
+    >
+      <span class="input-suffix">{{ denom | viewDenom }}</span>
+      <TmField id="amount" :value="rewards | atoms | fullDecimals" readonly />
+    </TmFormGroup>
+  </ActionModal>
+</template>
+
+<script>
+import { viewDenom, uatoms, atoms, fullDecimals } from "../../scripts/num.js"
+import ActionModal from "common/ActionModal"
+import TmField from "common/TmField"
+import TmFormGroup from "common/TmFormGroup"
+
+export default {
+  name: `modal-withdraw-rewards`,
+  components: {
+    ActionModal,
+    TmField,
+    TmFormGroup
+  },
+  filters: {
+    atoms,
+    viewDenom,
+    fullDecimals
+  },
+  props: {
+    validatorAddress: {
+      type: String,
+      required: true
+    },
+    rewards: {
```

Comment:
Thinking about the ActionModal webcomponent, does it make sense to make the rewards an input? Or would we wrap this in a controller?

Reply:
Yep, I thought about this. I believe wrapping it in a controller is better.

IMHO, the Modals should be 'dumb' views, and at most receive data and forward user actions, but that's it. Separation of concerns.

Reply:
(Note: This should apply to most views)